driver/utils/type_info.h

```diff
@@ -0,0 +1,2112 @@
+                    return fillOutputString<SQLCHAR>(src, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+                else if constexpr (is_string_data_source_type_v<SourceType>) {
+                    return fillOutputString<SQLCHAR>(src.value, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+                else {
+                    std::string dest_obj;
+                    to_null(dest_obj);
+                    ::value_manip::from_value<SourceType>::template to_value<std::string>::convert(src, dest_obj);
+                    return fillOutputString<SQLCHAR>(dest_obj, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+            }
+        };
+    };
+
+    template <>
+    struct to_buffer<SQLWCHAR *> {
+        using DestinationType = SQLWCHAR *;
+
+        template <typename SourceType>
+        struct from_value {
+            static inline SQLRETURN convert(const SourceType & src, BindingInfo & dest) {
+                if constexpr (std::is_same_v<SourceType, std::string>) {
+                    return fillOutputString<SQLWCHAR>(src, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+                else if constexpr (is_string_data_source_type_v<SourceType>) {
+                    return fillOutputString<SQLWCHAR>(src.value, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+                else {
+                    std::string dest_obj;
+                    to_null(dest_obj);
+                    ::value_manip::from_value<SourceType>::template to_value<std::string>::convert(src, dest_obj);
+                    return fillOutputString<SQLWCHAR>(dest_obj, dest.value, dest.value_max_size, dest.value_size, true);
+                }
+            }
+        };
+    };
+
+    template <>
+    struct to_buffer<SQL_NUMERIC_STRUCT> {
+        using DestinationType = SQL_NUMERIC_STRUCT;
+
+        template <typename SourceType>
+        struct from_value {
+            static inline SQLRETURN convert(const SourceType & src, BindingInfo & dest) {
+                if constexpr (std::is_same_v<SourceType, DestinationType>) {
+                    if (
+                        src.precision == dest.precision &&
+                        src.scale == dest.scale
+                    ) {
+                        return fillOutputPOD(src, dest.value, dest.value_size);
+                    }
+                }
+
+                SQL_NUMERIC_STRUCT dest_obj;
+                to_null(dest_obj);
+
+                dest_obj.precision = dest.precision;
+                dest_obj.scale = dest.scale;
+
+                ::value_manip::from_value<SourceType>::template to_value<DestinationType>::convert(src, dest_obj);
+
+                return fillOutputPOD(dest_obj, dest.value, dest.value_size);
+            }
+        };
+    };
+
+    template <typename SourceType>
+    struct from_buffer {
```

Comment:
`from_buffer<X>::to_value<Y>::convert()` => `convertFromBufferToValue(const X & x, Y & y)` ?

Reply:
That's severely affect maintainability, as mentioned earlier.